### PR TITLE
don't harass to unlock a paper key that's already unlocked

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: go/service/rekey_ui_handler.go [hunk 1]
```diff
@@ -111,11 +111,16 @@ func (r *RekeyUIHandler) rekeyNeeded(ctx context.Context, item gregor.Item) (err
...
 
-func keySolvesProblemTLF(key libkb.GenericKey, tlf keybase1.ProblemTLF) bool {
-	ourKid := key.GetKID()
-	for _, kid := range tlf.Solution_kids {
-		if kid.Equal(ourKid) {
-			return true
+func keysSolveProblemTLF(keys []libkb.GenericKey, tlf keybase1.ProblemTLF) bool {
+	var ourKIDs []keybase1.KID
+	for _, key := range keys {
+		ourKIDs = append(ourKIDs, key.GetKID())
```
Comment (patrickxb) [inline go/service/rekey_ui_handler.go]: key can be nil...only append if not

File: go/service/rekey_ui_handler.go [hunk 2]
```diff
@@ -135,14 +140,25 @@ func currentDeviceSolvesProblemSet(g *libkb.GlobalContext, ps keybase1.ProblemSe
...
+
+	// We can continue though, so no need to error out
+	if err != nil {
+		g.Log.Info("| Error getting paper key: %s\n", err)
+		err = nil
+	}
+
 	for _, tlf := range ps.Tlfs {
-		if !keySolvesProblemTLF(key, tlf) {
+		if !keysSolveProblemTLF([]libkb.GenericKey{deviceKey, paperKey}, tlf) {
```
Comment (patrickxb) [inline go/service/rekey_ui_handler.go]: leave paperKey out of array if nil?

**Comment by maxtaco** [comment]
fixed, thanks

**Comment by patrickxb** [comment]
Could have sworn I wrote "LGTM" this morning, but:  LGTM

**Comment by maxtaco** [comment]
Thanks for the review. This code will make it through the rewrite, I think.

**Pull request merged**